research/picai/prepare_data.py

```diff
@@ -0,0 +1,209 @@
+        annotation_base_filename = annotation_filename.split(".")[0]
+
+        # All of the scans for a given patient, even if there are multiple sets from different study ids,
+        # are stored in a single folder under patient id in the raw data.
+        patient_id, study_id = annotation_base_filename.split("_")
+        scans_path = os.path.join(scans_read_dir, patient_id)
+
+        # Get all the filenames that match the extension we expect and have the proper study id.
+        scan_filenames = [
+            path for path in os.listdir(scans_path) if study_id in path and path.endswith(scan_extension)
+        ]
+
+        # Extract the scans that we care about (t2w, adc, hbv) and store in a list ordered as we expect.
+        t2w_scan_filename = scan_filenames[[i for i, sf in enumerate(scan_filenames) if "t2w" in sf][0]]
+        adc_scan_path_filename = scan_filenames[[i for i, sf in enumerate(scan_filenames) if "adc" in sf][0]]
+        hbv_scan_path_filename = scan_filenames[[i for i, sf in enumerate(scan_filenames) if "hbv" in sf][0]]
+        ordered_scan_filenames = [t2w_scan_filename, adc_scan_path_filename, hbv_scan_path_filename]
+        scan_paths = [Path(os.path.join(scans_path, f)) for f in ordered_scan_filenames]
+
+        # Create sample and add to list of cases to be processed.
+        sample = PicaiCase(scan_paths, annotation_path, settings)
+        samples.append(sample)
+
+    # Preprocess list of samples, generate dataset overviews and write them to disk
+    paths_for_each_sample = preprocess(samples, DEFAULT_TRANSFORMS, num_threads=num_threads)
+    generate_dataset_json(paths_for_each_sample, overview_write_dir, splits_path)
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--scans_read_dir", type=str, required=True, help="Directory to read scans from.")
+    parser.add_argument("--annotation_read_dir", type=str, required=True, help="Directory to read annotation from.")
+    parser.add_argument("--scans_write_dir", type=str, required=True, help="Directory to write scans to.")
+    parser.add_argument("--annotation_write_dir", type=str, required=True, help="Directory to write annotation to.")
+    parser.add_argument(
+        "--overview_write_dir", type=str, required=True, help="Directory to write dataset overviews to."
+    )
+    parser.add_argument(
+        "--size",
+        type=int,
+        nargs="+",
+        required=False,
+        help="Size to convert images and annotations to. Default is to keep as is.",
+    )
+    parser.add_argument(
+        "--spacing",
+        type=float,
+        nargs="+",
+        required=False,
+        help="Spacing to convert images and annotations to. Default is to keep as is.",
+    )
+    parser.add_argument(
+        "--physical_size",
+        type=float,
+        nargs="+",
+        required=False,
+        help="Size to convert images and annotations to. Default is to keep as is.",
```

Comment:
Perhaps adding what the distinction is between the `size` and `physical_size` args would be helpful here?

Reply:
Added above in documentation for prepare data and will also add here!